pkg/config/runtime.go

```diff
@@ -105,19 +104,15 @@ func (r *Runtime) resolveRules(bag attribute.Bag, aspectSet AspectSet, rules []*
 
 // combined returns a Combined config given an aspect config
 func (r *Runtime) combined(aa *pb.Aspect, aspectSet AspectSet) *Combined {
-	if !aspectSet[aa.GetKind()] {
-		glog.V(3).Infof("Aspect Rejected %v not is set (%v)", aa.GetKind(), aspectSet)
+	if !aspectSet[aa.Kind] {
+		glog.V(3).Infof("Aspect rejected: %v not in set [%v]", aa.Kind, aspectSet)
 		return nil
 	}
 
 	var adp *pb.Adapter
 	// find matching adapter
 	// assume that config references are correct
-	if aa.GetAdapter() != "" {
-		adp = r.adapterByName[aa.GetAdapter()]
-	} else {
-		adp = r.adapterByKind[aa.GetKind()][0]
-	}
+	adp = r.adapterByName[aa.Adapter]
```

Comment:
aspect specifies kind or adapter.  If adapter is specified by should lookup the adapter by name. If no adapter name is specified we use the 1st adapter available of the correct kind.
```
  - kind: application-logs
    params:
      logName: "mixer_log"
      logEntryDescriptorNames: ["default"]
```
The above config should select the 1st adapter of kind = `application-logs`


```
  - name: stderr-logs
    kind: application-logs
    impl: stdioLogger
    params:
      logStream: 0 # STDERR
```

```
  - kind: application-logs
    adapter: stderr-logs
    params:
      logName: "mixer_log"
      logEntryDescriptorNames: ["default"]
```
This should specifically select the named adapter `stderr-logs`